### PR TITLE
Modern unix

### PR DESCRIPTION
**Pull request by candrewlee14**
PR for #1 
This will resolve:
- https://github.com/ibraheemdev/modern-unix/issues/99
- https://github.com/ibraheemdev/modern-unix/issues/5
- https://github.com/ibraheemdev/modern-unix/issues/85

## Pre-Requisites:
- [x] add group to bulk add and bulk remove packages (resolved https://github.com/candrewlee14/webman/issues/1)
## Packages
- [x] bat
- [x] exa
- [x] lsd
- [x] delta
- [x] dust
- [x] duf
- [x] broot
- [x] fd
- [x] rg
- [ ] ag _(* no web-linked distributed binaries)_
- [x] fzf
- [x] mcfly
- [x] choose
- [x] jq
- [x] sd
- [x] cheat
- [ ] tldr _(* no web-linked distributed binaries, can be installed with `npm`)_
- [x] bottom
- [ ] glances _(* no web-linked distributed binaries, can be installed with `pip`)_
- [ ] gtop _(* no web-linked distributed binaries, can be installed with `npm`)_
- [x] hyperfine
- [x] gping
- [x] procs
- [ ] httpie _(* no web-linked distributed binaries, can be installed with `pip`)_
- [x] curlie
- [x] xh
- [x] zoxide
- [x] dog

### CODE REVIEW

**Comment by candrewlee14** [comment]
`broot` (https://dystroy.org/broot/install/) has a tricky release format:
  - download link has linux and windows releases for only latest version
  - previous versions are zipped all together, not individually, on Github releases.

The options to handle this are:
1. add feature to unzip release bundle and select individual extracted folder
2. force latest download

I think forcing the latest download is probably my pick.
It's easy to implement and avoids wasting too much time on these one-off tricky cases.
There'd be an option added to the yaml file like `force_latest: true`, and the installation process would quit if the version argument is not the latest version.

**Comment by candrewlee14** [comment]
ag (https://github.com/ggreer/the_silver_searcher) is another tricky one. It doesn't really look like there are OS/Arch-specific downloads anywhere? Maybe I'm just not seeing them...

**Comment by candrewlee14** [comment]
choose (https://github.com/theryangeary/choose/) distributes an uncompressed binary directly. I need a way to handle that.

Edit: solved with latest webman commit adding is_binary option.

**Comment by AndrusGerman** [comment]
The Silver Searcher, @candrewlee14  maybe for these cases if you have to add a preparation label, 
Something that allows you to run commands if some packages need to be compiled.

```yaml
base_download_url: https://github.com/ggreer/the_silver_searcher/archive/refs/tags/2.2.0.tar.gz
requirement:
  - windows:
      - webman:mingw-ge # Example auto install webman package
  - linux:
      - system:gcc # Example alert please install external need package 

prepare_linux:
  - build step 1
  - build step 2
  - make
 
prepare_windows:
  - mingw-get install gcc
  - build step 2
  - make
```

**Comment by candrewlee14** [comment]
`cheat` has a strange format. On linux and darwin, it is a directly gzipped binary, but on Windows, it is a zipped folder that has a root `dist` folder that contains a `cheat-windows-amd64.exe`. Linux and darwin work correctly, and Windows installs but the binary link is `cheat-windows-amd64` when it probably should be `cheat`.
I could probably add a `bin_rename` dictionary field that matches a regex and renames accordingly...any other ideas?

**Comment by candrewlee14** [comment]
> The Silver Searcher, @candrewlee14 maybe for these cases if you have to add a preparation label, Something that allows you to run commands if some packages need to be compiled.
> 
> ```yaml
> base_download_url: https://github.com/ggreer/the_silver_searcher/archive/refs/tags/2.2.0.tar.gz
> requirement:
>   - windows:
>       - webman:mingw-ge # Example auto install webman package
>   - linux:
>       - system:gcc # Example alert please install external need package 
> 
> prepare_linux:
>   - build step 1
>   - build step 2
>   - make
>  
> prepare_windows:
>   - mingw-get install gcc
>   - build step 2
>   - make
> ```

@AndrusGerman Hmm, that's an interesting idea. 
For security reasons and simplicity, I'd prefer to not allow package recipes to run commands. 
I think if a package doesn't distribute a binary for a specific arch & platform, `webman` should not attempt to build from source. I may revisit this later, but for now, it'll be better to focus on building recipes for binaries that do exist.

**Comment by candrewlee14** [comment]
`glances` has no web-linked distributed binaries. It is easily installable with `pip` though. This is a similar situation as with `ag`.

Now the question is: do we support running commands to install binaries (with dependencies), OR do not support running arbitrary package installation commands.

For now, I'm going to choose the latter. I think for security and simplicity, it's better to just support packages with distributed binaries. This makes sure that `exec()` is not called for any package script. It also means that we can guarantee all the installed binaries are located inside the webmanDir (default: `~/.webman`).

**Comment by candrewlee14** [comment]
The majority of the packages are added, and the ones that aren't had no binaries but are installable with language-specific package managers. They also have use cases that are filled by other packages here:
- `tldr`: use `cheat`
- `httpie`: use `curlie` or `xh`
- `glances` & `gtop`: use `bottom`

**Comment by candrewlee14** [comment]
The majority of the packages are added, and the ones that aren't had no binaries but are installable with language-specific package managers. They also have use cases that are filled by other packages here:

tldr: use cheat
httpie: use curlie or xh
glances & gtop: use bottom
